lib/autoload.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+class ClassLoader {
+	private $searchPath = [];
+
+	/**
+	 * Register path in which to look for a class.
+	 */
+	public function registerPath($path) {
+		if (is_string($path)) {
+			$this->searchPath[] = $path;
+		}
+		if (is_array($path)) {
+			array_push($this->searchPath, ...$path);
+		}
+	}
+
+	/**
+	 * Load class file if found.
+	 */
+	public function loadClass($class)
+	{
+		if ($file = $this->findFile($class)) {
+			require $file;
+		}
+	}
+
+	/**
+	 * Find the file containing the class definition.
+	 */
+	public function findFile($class) {
```

Comment:
I am not so found of *searching*  for the class on disk, when we have the possibility of directly knowing the class filename 😟

Reply:
I agree. The initial problem was that it was too restricted and too dependent on the structure. It is still but in a different way.
Before, the rule was really strict on how to name the classes (ex: must end with _Controller), which give weird convention name and also discrepancies (ex: `FreshRSS_api_Controller` vs `FreshRSS_AlreadySubscribed_Exception`, search in Controllers for controllers but in Model for class with only one word after freshrss). That make no sense to me.
Now, there is only one rule, the class name has to match its file.

The idea with the namespace was to be able to match directly the file by using the namespace instead of relying on a weird convention on the class name to find the file to include. If you compare the process of including the class `Controller\ApiController` versus the class `FreshRSS_api_Controller`, it will be easier to convert to a path the later than the former.
With that in mind, we could remove most of the disk access to load the classes because we now right away where it is without relying on some convoluted method to extract the path from the class.